### PR TITLE
#53 ADD docs for utils modules

### PR DESCRIPTION
**Pull request by jalew188**

### CODE REVIEW

File: alpharaw/utils/centroiding.py [hunk 1]
```diff
@@ -28,7 +46,13 @@ def naive_centroid(
 
 
 @njit
-def _find_sister_peaks(peak_mzs, centroiding_peak_tols, start):
+def _find_sister_peaks(
+    peak_mzs: np.ndarray, centroiding_peak_tols: np.ndarray, start: int
+):
+    """
+    Find sister peak stop idx for the given start idx.
```
Comment (mschwoer) [inline alpharaw/utils/centroiding.py:53]: to me, "Sister Peak" sounds like a colleague of Nurse Ratched :-D
mb introduce this concept here?

File: alpharaw/utils/ms_path_utils.py [hunk 2]
```diff
@@ -14,10 +14,23 @@
...
+
+    Returns
+    -------
+    str
+        The raw name without extension.
+    """
     raw_name = os.path.basename(ms_file)
     lower_name = raw_name.lower()
-    for _ext in special_ms_exts:
+    for _ext in _special_ms_exts:
```
Comment (mschwoer) [inline alpharaw/utils/ms_path_utils.py:33]: could you make this constant "_ALL_UPPER_CASE"?

File: alpharaw/utils/ms_path_utils.py [hunk 3]
```diff
@@ -14,10 +14,23 @@
 ]
 
 
-def get_raw_name(ms_file: str, special_ms_exts: list = _special_ms_exts):
+def get_raw_name(ms_file: str) -> str:
+    """
+    Get raw name str from the ms file path.
+
```
Comment (mschwoer) [inline alpharaw/utils/ms_path_utils.py:20]:
```suggestion
    Get base name of raw data file or folder from the file path by removing the first extension defined in `_SPECIAL_MS_EXTS`
    Example:
    >>> get_raw_name("/path/to/raw_data1.d") = "raw_data1"
```

File: alpharaw/utils/ms_path_utils.py [hunk 4]
```diff
@@ -14,10 +14,23 @@
...
+    str
+        The raw name without extension.
+    """
     raw_name = os.path.basename(ms_file)
     lower_name = raw_name.lower()
-    for _ext in special_ms_exts:
+    for _ext in _special_ms_exts:
         if lower_name.endswith(_ext.lower()):
             raw_name = raw_name[: -len(_ext)]
             break
```
Comment (mschwoer) [inline alpharaw/utils/ms_path_utils.py:36]: is the logic in l. 24-25 required? as far as I get it, this method removes all extensions, not matter if they are defined in `_special_ms_exts` or not?

**Review by mschwoer — APPROVED** [review]

File: alpharaw/utils/ms_path_utils.py (hunk 3 above)
Comment (jalew188) [reply alpharaw/utils/ms_path_utils.py:20]: Yep

File: alpharaw/utils/ms_path_utils.py (hunk 4 above)
Comment (jalew188) [reply alpharaw/utils/ms_path_utils.py:36]: This is required as os.path.splitext cannot recognize `raw_name.d.hdf` this kind of extension formats...

Comment (mschwoer) [reply alpharaw/utils/ms_path_utils.py:36]: alright, but still l.24-25 remove all extensions .. is this desired? or should we limit this to `[.d, .raw, ..]`

Comment (jalew188) [reply alpharaw/utils/ms_path_utils.py:36]: no, for instance, alphatims save the bruker file as `raw_name.d.hdf`. And as HDF is a general file format, so if we use `*.hdf`, then we don't know if this is a MS data or not. So `*.raw.hdf` or `*.wiff.hdf` is preferred.

File: alpharaw/utils/centroiding.py (hunk 1 above)
Comment (jalew188) [reply alpharaw/utils/centroiding.py:53]: Done

File: alpharaw/utils/ms_path_utils.py (hunk 4 above)
Comment (mschwoer) [reply alpharaw/utils/ms_path_utils.py:43]: this is a good example where a unit test would help to gather and implicitly document all the usecases and the desired behaviour of this method :-)

Comment (jalew188) [reply alpharaw/utils/ms_path_utils.py:43]: That's right

**Pull request merged**